File: javascript/two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    hash = {}

    for (var i = 0; i < nums.length; ++i)
    {
    	if (target - nums[i] in hash)
    	{
    		return [hash[target - nums[i]] + 1, i + 1];
    	}

    	hash[nums[i]] = i;
    }
};

numbers = [2, 7, 11, 15];
console.log(twoSum(numbers, 9));

File: ruby/two_sum.rb
# Given an array of integers, find two numbers such that they add up to a specific target number.

# The function twoSum should return indices of the two numbers such that they add up to the target, where index1 must be less than index2. Please note that your returned answers (both index1 and index2) are not zero-based.

# You may assume that each input would have exactly one solution.

# Input: numbers={2, 7, 11, 15}, target=9
# Output: index1=1, index2=2

# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}

def two_sum(nums, target)
	hash = {}

	nums.each_with_index do |x, index|
		return [hash[target - x] + 1, index + 1] if hash.has_key? target - x
		hash[x] = index;
	end

	[]
end